CLN: Drop compact_ints/use_unsigned from read_csv

gfyoung opened the pull request:
Deprecated in v0.19.0

xref #13323

gfyoung committed:
CLN: Drop compact_ints/use_unsigned from read_csv

Deprecated in v0.19.0

xref gh-13323

gfyoung commented:
All is green.  Anyone welcome to have a look to merge this.

jreback commented:
wait on merging anything

gfyoung commented:
Ah, right.  Travis is still failing.  Got it.

jreback commented on pandas/io/parsers.py:
```diff
@@ -1569,11 +1544,6 @@ def _convert_to_ndarrays(self, dct, na_values, na_fvalues, verbose=False,
                 if cast_type and not is_dtype_equal(cvals, cast_type):
                     cvals = self._cast_types(cvals, cast_type, c)
 
-            if issubclass(cvals.dtype.type, np.integer) and self.compact_ints:
```
I believe we can also blow away ``lib.downcast_int64``, ``to_numeric`` uses a different (and better) mechanism for doing this. IIRC ``downcast_int64`` was only for compatibility.

gfyoung replied:
GitHub search seems to verify that belief.

gfyoung committed:
CLN: Remove downcast_int64 from inference.pyx

It was only being used for the compact_ints
and use_unsigned parameters in read_csv.

gfyoung commented:
Function removed, and all is green still.  PTAL.

jreback commented:
thanks @gfyoung
